src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -1149,6 +1162,124 @@ public bool PutCounters(DocumentsOperationContext context, string documentId, st
             }
         }
 
+        public int FixCountersForDocument(DocumentsOperationContext context, string documentId)
+        {
+            List<string> allNames = null;
+            LazyStringValue collection = default;
+
+            Table writeTable = default;
+            int numOfCounterGroupFixed = 0;
+            CollectionName collectionName = default;
+            try
+            {
+                var table = new Table(CountersSchema, context.Transaction.InnerTransaction);
+
+                using (DocumentIdWorker.GetSliceFromId(context, documentId, out Slice key, separator: SpecialChars.RecordSeparator))
+                {
+                    foreach (var result in table.SeekByPrimaryKeyPrefix(key, Slices.Empty, 0))
+                    {
+                        var tvr = result.Value.Reader;
+                        BlittableJsonReaderObject data;
+
+                        using (data = GetCounterValuesData(context, ref tvr))
+                        {
+                            data = data.Clone(context);
+                        }
+
+                        data.TryGet(Values, out BlittableJsonReaderObject counterValues);
+                        data.TryGet(CounterNames, out BlittableJsonReaderObject counterNames);
+                        if (counterValues == null || counterNames == null)
+                            throw new InvalidDataException("Invalid CounterGroup data - missing counter values or counter names");
```

Comment:
when can this happened? if we hit this, we will not fix anything further, right?

Reply:
removed